types/execution/execution-triple.ts

```diff
@@ -0,0 +1,55 @@
+import {InstructionSet} from '../instructionsets.js';
+
+export enum ExecutionSpecialty {
+    cpu = 'cpu',
+    nvgpu = 'nvgpu',
+    amdgpu = 'amdgpu',
+}
+
+export class BaseExecutionTriple {
```

Comment:
What makes this file something that lives in `types` ? It seems like a `class` object with behaviour etc?

Reply:
I was considering this being available on frontend to see if remote execution is available, but in the end I realized thats not possible.